galley/pkg/runtime/projections/serviceentry/handler.go

```diff
@@ -251,7 +265,7 @@ func (p *Handler) deleteMcpResource(fullName resource.FullName) {
 
 func (p *Handler) updateVersion() {
 	p.version++
-	monitoring.RecordStateTypeCount(collection.String(), len(p.mcpResources))
+	monitoring.RecordStateTypeCountWithContext(statsCtx, len(p.mcpResources))
```

Comment:
Seems like you can hang the statsCtx to handler?

Reply:
It's already a global ... is there a reason to change it?

Reply:
Unless I am missing something statsCtx is a global you're introducing with your latest CL.

Reply:
correct ... I'm just asking if there's a reason that it's incorrect as a global?

Reply:
It generally comes back to bite you? Static mutables are inherently problematic. (and context is mutable). We already ran into a couple issues with service restart not playing well with statics in dependency libraries, which causes issues with testing at the very least. Not adding statics helps and avoids refactoring pitfalls.

Reply:
changed to a member of Handler